### PR TITLE
WIP: add API for Cronjob

### PR DESCRIPTION
**Pull request by f-f**

### CODE REVIEW

**Comment by NeQuissimus** [comment]
@f-f Are you still working on this? :)

**Comment by f-f** [comment]
Hey @NeQuissimus!

Basically it went like this:
- we started porting our Kube Cronjobs to Dhall at work
- I started making the `api` modules based on that
- we ended up with much more complexity in our port, so it was not suitable anymore to use the api modules from here
- I restructured the types in a more comfortable way (I have some unpushed work in progress about that), and I'd like to cherry-pick some of the work done in `master`

TL;DR: this thing turned in a direction I don't like, so I think I'll open an alternative PR to replace it, and close this one
